### PR TITLE
Prevent stackoverflow errors to kill UdpMessageChannel thread

### PR DESCRIPTION
**Pull request by jpgiroux**
We have enconter a stackoverflow error originating from UdpMessageChannel killing the thread in the process making our application no longuer able to process packet. We want to make sure that this error does not kill the thread in the future to avoid down time.

### CODE REVIEW

File: src/gov/nist/javax/sip/stack/UDPMessageChannel.java [hunk 1]
```diff
@@ -318,6 +318,9 @@ public void run() {
 
                 logger.logError(
                         "Error while processing incoming UDP packet" + Arrays.toString(packet.getData()), e);
+            } catch (StackOverflowError e) {
+                logger.logError("StackOverflowError while processing incoming UDP packet" + Arrays.toString(packet.getData()));
```
Comment (jagliot) [inline src/gov/nist/javax/sip/stack/UDPMessageChannel.java]: Why not add the exception at the end of the log line like line 320? and delete the double log

File: src/gov/nist/javax/sip/stack/UDPMessageChannel.java [hunk 2]
```diff
@@ -318,6 +318,9 @@ public void run() {
```
Comment (jagliot) [inline src/gov/nist/javax/sip/stack/UDPMessageChannel.java:318]: you could also modify the catch exception to catch throwable and be done with it. 1 line fix.

File: src/gov/nist/javax/sip/stack/UDPMessageChannel.java (hunk 1 above)
Comment (jpgiroux) [reply src/gov/nist/javax/sip/stack/UDPMessageChannel.java]: The reason is that UDPMessageChannel is not using a typical `java.util.logging.Logger` but a `gov.nist.core.StackLogger`. This logger API is not as flexible. 

Here:

image

Comment (jpgiroux) [reply src/gov/nist/javax/sip/stack/UDPMessageChannel.java]: I have refactor the `logError(String, Exception): void` for `logError(String, Throwable): void` so now the catch clause is cleaner.

File: src/gov/nist/javax/sip/stack/UDPMessageChannel.java (hunk 2 above)
Comment (jpgiroux) [reply src/gov/nist/javax/sip/stack/UDPMessageChannel.java:318]: done!

**Pull request merged**